modules/k8s/namespace_test.go

```diff
@@ -34,3 +35,50 @@ func TestNamespaces(t *testing.T) {
 	namespace := GetNamespace(t, options, namespaceName)
 	require.Equal(t, namespace.Name, namespaceName)
 }
+
+func TestNamespaceWithMetadataE(t *testing.T) {
+	t.Parallel()
+
+	uniqueId := random.UniqueId()
+	namespaceName := strings.ToLower(uniqueId)
+	options := NewKubectlOptions("", "", namespaceName)
+	namespaceLabels := map[string]string{"foo": "bar"}
+	namespaceObjectMetaWithLabels := metav1.ObjectMeta{
+		Name:   namespaceName,
+		Labels: namespaceLabels,
+	}
+	err := CreateNamespaceWithMetadataE(t, options, namespaceObjectMetaWithLabels)
+	require.NoError(t, err)
+	defer func() {
+		DeleteNamespace(t, options, namespaceName)
+		namespace := GetNamespace(t, options, namespaceName)
+		require.Equal(t, namespace.Status.Phase, corev1.NamespaceTerminating)
+	}()
+
+	namespace := GetNamespace(t, options, namespaceName)
+	require.Equal(t, namespace.Name, namespaceName)
+	require.Equal(t, namespace.Labels, namespaceLabels)
+}
+
+func TestNamespaceWithMetadata(t *testing.T) {
```

Comment:
If you update `CreateNamespaceWithMetadata` to call `CreateNamespaceWithMetadataE` under the hood, then you only need one test for `CreateNamespaceWithMetadata`, and it'll cover both of the functions.

Reply:
yep deleted this test! 👌